File: FRONT-END/lib/dto/app/auth/response/recaptcha_response_dto.dart
class RecaptchaResponseDTO {
  final bool success;
  final String? challengeTs;
  final String? hostname;
  final List<String>? errorCodes;

  RecaptchaResponseDTO({
    required this.success,
    this.challengeTs,
    this.hostname,
    this.errorCodes,
  });

  factory RecaptchaResponseDTO.fromJson(Map<String, dynamic> json) {
    return RecaptchaResponseDTO(
      success: json['success'] ?? false,
      challengeTs: json['challenge_ts'],
      hostname: json['hostname'],
      errorCodes: json['error-codes'] != null 
          ? List<String>.from(json['error-codes'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'success': success,
      'challenge_ts': challengeTs,
      'hostname': hostname,
      'error-codes': errorCodes,
    };
  }
}

File: BACK-END/KuentecoApp/src/main/java/org/kuenteco/backend/dto/auth/RecaptchaResponseDTO.java
package org.kuenteco.backend.dto.auth;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;
import lombok.*;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RecaptchaResponseDTO {
    private boolean success;

    @JsonProperty("challenge_ts")
    private String challenge_ts;

    private String hostname;

    @JsonProperty("error-codes")
    private List<String> errorCodes;
}
